### ISSUE TITLE
Incorrect test results for Feature and Scenario names that were changed during runtime

### ISSUE THREAD

**Issue by iamkenos** [hrcorval/behavex#156]
**Describe the bug**
almost all of our tests in our repo are very much data driven. we have a ton of feature files with example tables and we want the unique columns of the example tables appended in the scenario names.

e.g.
![image](https://github.com/user-attachments/assets/50d516c1-332a-499d-855b-cc3bd5bf729e)

however, we dont want to add this on every scenario in our feature files since we know that the examples are always going to have this column.

`behave` exposes the ability to mutate feature and scenario names during runtime via context hooks. we took advantage of this and mutated the scenario names on such hooks, as seen below

```python
# environment.py

import os
from dotenv import load_dotenv

def before_all(context):
    load_dotenv()

def before_feature(context, feature):
    # change the feature name on runtime
    feature.name = f"[{os.environ['BRAND']}] {feature.name}"

def after_scenario(context, feature):
    # change the scenario name on runtime
    data_id = 'PULLED_FROM_SOMEWHERE'
    context.scenario.name = f'{context.scenario.name}: {data_id}'

    if ('@1.1' in context.scenario.name):
        raise AssertionError
```

this used to work in version `1.6` of behavex but is broken in the latest version.

**To Reproduce**
1. clone branch `bug/mutated-scenario-names-not-appearing-in-the-report` [this repo](https://github.com/iamkenos/behavex-sandbox/)
2. install deps
3. run the command `behavex -t one -t scenario --parallel-processes 2 --parallel-scheme scenario`

**Expected behavior**
- results are show as expected on the summary reporter and html reporter
- summary: (as taken from a `behave` run)
  - ![image](https://github.com/user-attachments/assets/be4e11a2-111c-4dcf-8be8-d2eb92da2b03)
- html
  - ![image](https://github.com/user-attachments/assets/2a101f00-96d5-458f-8806-8992f0967278)

**Current behavior**
- summary results are incorrect
  - ![image](https://github.com/user-attachments/assets/1672855d-f41b-4d11-ab96-d49062907190)
- HTML reporter results are incorrect
- Notice even the prefix on the feature title was removed as well.
  - ![image](https://github.com/user-attachments/assets/935624cc-be0d-4577-a7b8-eff1c35c1b2f)

I'd be happy to help with the fix if you can guide me around which files to look at. otherwise thanks again for all your work!

**Comment by hrcorval** [comment]
@iamkenos, very good catch. This is something we have seen before, but we didn't have time to fix yet. I believe the problem is that we are trying to execute and match the scenarios by name, so when the name changes we got a mismatch. 
Most probably, the solution would be to map the scenarios by the scenario line, which is already supported by Behave. By example:
`features/sample.feature:45`

In order to do it, we should skip sending the scenario names across the functions in the runner.py module, and also avoid triggering Behave (inside processes) with the scenario name (and start using the scenario line)
If you consider you can move forward with that change, I can help/support on whatever is needed.

Thanks

**Comment by iamkenos** [comment]
actually, behave (and behavex) is running the scenarios just fine. it's just the reporters (html and summary) that aren't showing the results. but the tests actually runs.

**Comment by hrcorval** [comment]
Hi @iamkenos . We have a fix already implemented in the following release candidate:
https://test.pypi.org/project/behavex/4.0.8rc1/

We have tested that one and works on our end. The changes are are available in the following commit:
https://github.com/hrcorval/behavex/pull/164/commits/4aac87893206fa01ac235c006b44f02673252303

In case you are using pipenv, you can install the release candidate version using the following command:
`pipenv install -i https://test.pypi.org/simple/ behavex==4.0.8rc1`

I hope it helps on solving the reported issue

**Comment by hrcorval** [comment]
We have identified other areas for improvements, so we will come with a new RC soon, as we need additional improvements when running in parallel. Keep you posted on that.

**Comment by iamkenos** [comment]
hi @hrcorval - thank for working on the fix! i'm testing this via symlinking the latest `release_4.0.8` to my local repo @ commit `f93a353`. 
![image](https://github.com/user-attachments/assets/71970133-074d-4751-9eb6-b08ad396dfde)

im still getting the issue, however.
![image](https://github.com/user-attachments/assets/7019be04-e39a-4e17-b1d5-25521b1522ec)
![image](https://github.com/user-attachments/assets/45b7a19b-f084-4b0d-afd2-a078bcfa9782)


lemme know how i can help!

**Comment by iamkenos** [comment]
hi @hrcorval , good day! i tested this against the latest commit of release `4.0.8` branch. looks good now.

**Comment by hrcorval** [comment]
Hi @iamkenos, thanks for confirming this. We completed the changes to manage test scenarios by location (scenario line) instead of scenario names, and this solved the issue that arises when dynamically changing the scenario names. 
We released the new library version 4.0.8 that is ready to install in pypi. 
We hope it helps! :)
